src/main/java/com/google/sps/servlets/AddOrganizationServlet.java

```diff
@@ -52,15 +60,31 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     openHours.add(Integer.parseInt(hourOpen));
     openHours.add(Integer.parseInt(hourClosed));
 
-    ArrayList<String> moderatorList = new ArrayList<String>();
-    moderatorList.add("anon creator"); //dummy value for before auth gets set up
+    /* This implementation stores history entries as embedded entities instead of custom objects
+     * because it is much simpler that way */
+    ArrayList changeHistory = new ArrayList<>();
+    UserService userService = UserServiceFactory.getUserService();
+    boolean isUserLoggedIn = userService.isUserLoggedIn();
+    String username;
+    if (isUserLoggedIn) {
```

Comment:
Should you allow an organization edit if the user is not logged in?

Reply:
My first implementation just allowed this, but this new commit throws an IllegalArgumentException instead to protect the database from unknown users

Reply:
If the entire function has the same access level, it'd be better to perform the access check at the start of the function, s.t. as the function evolves over time it stays secure. Security here means items ranging from the obvious: another transaction to more subtle: like timing attacks